### DISCUSSION TITLE
PostCSS plugins configuration

### DISCUSSION THREAD

**Discussion by hugo-t-b** [nuxt/nuxt#16870] [General]
I am using postcss-jit-props in my Nuxt 3 app. It is working with this configuration:

```ts
build: {
    postcss: {
      postcssOptions: {
        plugins: {
          "postcss-jit-props": openProps
        }
      }
    }
 }
```

However, configuring PostCSS this way is very cumbersome. I am wondering if there is a better way to configure PostCSS plugins.

**Comment by danielroe** [comment]
There is a new top level postcss option - see https://github.com/nuxt/framework/discussions/2883.

**Reply by baixiaoyu2997** [reply]
hi, danielroe,  dose nuxt3 now support [experimental](https://github.com/csstools/postcss-plugins/tree/main/experimental) plugin?
in doc ,need config plugin like this:
```
plugins: [
	// other plugins
	postcssPresetEnv({
		features: {
			'css-has-pseudo': false
		}
	}),
	// other plugins
	cssHasPseudoExperimental(), // last
]

```
how do this in nuxt3?

**Reply by danielroe** [reply]
Have you tried following the info at https://v3.nuxtjs.org/api/configuration/nuxt.config#postcss?

**Reply by ncryer** [reply]
@danielroe can you provide any additional insight to how postcss plugins are supposed to work in Nuxt3? The documentation is very sparse.
 
I have not been able to get the following mini-example to work despite hours of trying (example borrowed from https://tailwindcss.nuxtjs.org/examples/postcss-plugin/): 
 
`App.vue`: 
```vue
<template>
  <div class="test">
    PostCSS <span class="test_gray">Color Gray</span> is
    <span class="test_working">working</span>!
  </div>
</template>

<style lang="postcss">
.test {
  color: rgb(40, 176, 218);
  @apply p-4;

  &_working {
    @apply text-green-500;
  }

  &_gray {
    color: gray(50);
  }
}
</style>
```
`nuxt.config.ts`: 
```typescript 
export default defineNuxtConfig({
    postcss: {
        plugins: {
            'postcss-color-gray': {},
        }
    }
})
```
I have attempted many permutations of Vue2 and Vue3 config examples to no avail. How is this supposed to work?

**Reply by danielroe** [reply]
That is how the configuration syntax is meant to be. Maybe you can provide a mini sandbox and say what you are expecting instead?

**Reply by ncryer** [reply]
@danielroe thanks so much for taking a look at this! Here's a stackblitz: https://stackblitz.com/edit/nuxt-starter-oxk7du?file=app.vue 

The expected output is having several of the words colored

**Reply by ncryer** [reply]
@danielroe did you get a chance to take a look at this? Should I create a separate issue?

**Reply by danielroe** [reply]
Apologies! Yes, please, an issue will mean I don't forget.
